examples/ex18.cpp

```diff
@@ -89,6 +90,8 @@ int main(int argc, char *argv[])
                   "Number of times to refine the mesh uniformly.");
    args.AddOption(&order, "-o", "--order",
                   "Order (degree) of the finite elements.");
+   args.AddOption(&num_flux_type, "-nf", "--numerical-flux",
+                  "Numerical flux: 1 - Rusanov, 2 - Godunov (component-wise).");
```

Comment:
```suggestion
                  "Numerical flux: 1 - Rusanov, 2 - Godunov (using scalar equation Riemann solver).");
```

Reply:
I do not find this description very clear honestly. First, it is not clear how something scalar is applied to vectors. Secondly, Godunov is an approximate Riemann solver itself, so it is confusing what is using what. While "component-wise" says it relatively clearly within the limited space. Maybe "component-wise scalar"?

Reply:
How about `Scalar Godunov (component-wise)`?

Reply:
>Godunov is an approximate Riemann solver itself, so it is confusing what is using what. 

No, the Godunov method can use either an exact Riemann solution or an approximate Riemann solution. As I've noted in other comments, consider the Godunov method not as Riemann solver (in that it does not provide a Riemann solution) but instead the Godunov method **uses** a Riemann solution (typically provided by a Riemann solver) to provide a numerical flux. Other methods exist to provide a numerical flux given a Riemann solution, e.g., the wave propagation algorithm by LeVeque.

> it is not clear how something scalar is applied to vectors. 

I agree and would approve not using here for the Euler equations

> Maybe "component-wise scalar"?

If your suggestion is "2 - Godunov (component-wise scalar)", then that works for me.

Reply:
I'll note that whatever change is decided on, ex18 should be updated to match.

Reply:
Apologies for any confusion from emails: I had accidentally left comments "pending" that required some cleanup.

> Godunov is an approximate Riemann solver itself

No, the Godunov method can use an exact Riemann solution or an approximate Riemann solution.  As I've noted elsewhere, consider the Godunov method to **use** a Riemann solution / solver to provide a numerical flux instead of being a Riemann solver (i.e., the Godunov method itself does not supply a Riemann solution and there are other methods for providing a numerical flux from a Riemann solution, e.g., the wave propagation algorithm by LeVeque). Such a view should clear up the confusion on what is using what.

> Maybe "component-wise scalar"?

If the suggestion is "2 - Godunov (component-wise scalar)", then that would address the original concern behind this thread.  Given, as you noted, it is not clear that applying something scalar to a vector is valid (and in this case, I don't think it's valid to use a scalar Riemann solution for the Euler equations), I would also support not providing this option at all.

Reply:
As I see it, the current framework is structured such way that `NumericalFlux` (formerly `RiemannSolver`) is based on:

- approximate Riemann solver - Both currently implemented classes usable with any `FluxFunction` under some necessary assumptions on monotonicity/convexity.
- exact Riemann solver - Could be some class directly tailored to some problem and solving the Riemann problem analytically, which would _not_ take any `FluxFunction` as input, because some logic for the flux would have to be "baked" into the class, going beyond the interface of a `FluxFunction`. Currently not implemented.

Anyway, it seems we have the result 2 vs. 1 reviewer about example 18. It was not my original intention anyway, so I will remove it. When we implement systems, we may return to this switching code.

Reply:
Removed in https://github.com/mfem/mfem/pull/4513/commits/9943072dcaba138792ccb3f0962a042b33f828e2 .

Reply:
As I see it, there are numerical fluxes that do not use Riemann solutions at all, e.g., the Rusanov / Local Lax Friedrichs flux.